gef.py

```diff
@@ -9850,44 +9641,44 @@ def do_invoke(self, argv):
             depth = 1
 
         try:
-            loc_start   = current_arch.pc
+            loc_start   = gef.arch.pc
             loc_end     = parse_address(argv[0])
         except gdb.error as e:
-            err("Invalid location: {:s}".format(e))
+            err(f"Invalid location: {e}")
             return
 
         self.trace(loc_start, loc_end, depth)
         return
 
-    def get_frames_size(self):
+    def get_frames_size(self) -> int:
         n = 0
         f = gdb.newest_frame()
         while f:
             n += 1
             f = f.older()
         return n
 
-    def trace(self, loc_start, loc_end, depth):
-        info("Tracing from {:#x} to {:#x} (max depth={:d})".format(loc_start, loc_end, depth))
-        logfile = "{:s}{:#x}-{:#x}.txt".format(self.get_setting("tracefile_prefix"), loc_start, loc_end)
+    def trace(self, loc_start: int, loc_end: int, depth: int) -> None:
+        info(f"Tracing from {loc_start:#x} to {loc_end:#x} (max depth={depth:d})")
+        logfile = f"{self['tracefile_prefix']}{loc_start:#x}-{loc_end:#x}.txt"
         enable_redirect_output(to_file=logfile)
         hide_context()
         self.start_tracing(loc_start, loc_end, depth)
         unhide_context()
         disable_redirect_output()
-        ok("Done, logfile stored as '{:s}'".format(logfile))
+        ok(f"Done, logfile stored as '{logfile}'")
         info("Hint: import logfile with `ida_color_gdb_trace.py` script in IDA to visualize path")
         return
 
-    def start_tracing(self, loc_start, loc_end, depth):
+    def start_tracing(self, loc_start: int, loc_end: int, depth: int) -> None:
         loc_cur = loc_start
         frame_count_init = self.get_frames_size()
 
         gef_print("#")
```

Comment:
for the project: Maybe we could make `gef_print` variadic so that we could just comma separate each of these lines into one call.

Reply:
#788